feature_engine/datetime/date.py

```diff
@@ -0,0 +1,169 @@
+# Authors: dodoarg <eargiolas96@gmail.com>
+
+from typing import List, Optional, Union
+
+import numpy as np
+import pandas as pd
+
+from feature_engine.dataframe_checks import _is_dataframe
+from feature_engine.datetime.base_transformer import DateTimeBaseTransformer
+from feature_engine.variable_manipulation import _check_input_parameter_variables
+
+
+class ExtractDateFeatures(DateTimeBaseTransformer):
+    """
+    ExtractDateFeatures extract various date-related features from
+    datetime features present in the dataset, adding new columns
+    accordingly.
+
+    **Notes**
+    Day of the month extraction isn't supported by panda modules.
+    This transformer implements a workaround that counts how many blocks
+    of 7 days have passed in a given month.
+    A more serious approach based on the calendar should be implemented
+    ***
+
+    Parameters
+    ----------
+    variables: list, default=None
+        The list of variables to impute. If None, the imputer will find and
+        select all datetime variables, including those converted from
+        features of type object/category.
+
+    features_to_extract: list, default = "year"
+        The list of date features to extract. See supported attribute
+
+    drop_datetime: bool, default = "True"
+        Whether to drop datetime variables from the dataframe, including
+        those that have been converted by the transformer.
+        Note: if you pass a subset of features via the variables argument
+        of the transformer and drop_datetime=True, there might be more
+        datetime features in the dataframe that will not be dropped
+        upon calling the transform method
+
+    kwargs:
+        see pd.to_datetime keyword arguments other than "errors"
+
+    Attributes
+    ----------
+    supported:
+        list of supported date features
+
+    Available upon calling the fit method:
+
+    variables_:
+        List of variables from which to extract date features
+
+    n_features_in_:
+        The number of features in the train set used in fit.
+
+    Methods
+    -------
+    fit:
+        Find the variables from which to extract date features
+    transform:
+        Add the new date features specified in features_to_extract argument
+    fit_transform:
+        Fit to the data, then transform it.
+    """
+
+    def __init__(
+        self,
+        variables: Union[None, int, str, List[Union[str, int]]] = None,
+        features_to_extract: Union[str, List[str]] = "year",
```

Comment:
I would default this to None, in which case we would extract all features. 

Alternatively, the user can only pass a list with the features to extract. Even if it is only 1 feature, we should allow only lists. Makes our lives easier, and for the user is not a massive headache to put the string in a list :p

Reply:
You want it to default to all features? When I was coding I had this design choice to make and I thought 7+ columns to add to the dataset __per variable__ would be a lot, that's why I thought of the two options "all" and None where None is default and returns one or a few (most relevant) features.
But I see your point, I guess it makes the code easier

Reply:
The thing is that the most relevant will change with the use case. If it is something recorded every minute, then year does not make much sense. And if it is something recorded monthly, then daily would not be appropriately. I know, the feature space may explode. But we hope the user is on top of what they need and can narrow down the selection? not sure.

Reply:
Yea that's why I wasn't sure about what to have it default it to. Maybe we make it a positional argument of the __init__ so that the user __has to__ specify what features to extract?

Especially if we add all those binary attributes you mentioned in another comment like is_month_start etc the feature_space is really gonna explode with something like 15+ new columns per variable

Reply:
Scikit-learn encourages us to have all parameters with default values, so that users can implement it out of the box. So if we can, let's stick to that.

The list of variables to get from time is quite extensive, I agree with you on the explosion of the feature space, so maybe, instead of outputting just one variable like "year", or all variables, the default could be to output the group that we consider the most common.

Maybe we can return by default year, month, day (1-31) and day of the week (0-6)?

So the parameter would take 3 types of input:

- `None`: returns the above 4 features
-` "all"`: returns all features
- `list:` with short names to the features the user wants returned.

What do you think?